src/api/app/controllers/staging_project_controller.rb

```diff
@@ -1,10 +1,11 @@
 class StagingProjectController < ApplicationController
-  before_action :set_project
+  before_action :require_login
+  before_action :set_staging_project
 
   private
 
-  def set_project
-    @project = Project.find_by(name: params[:project])
-    raise ActiveRecord::RecordNotFound unless @project
+  def set_staging_project
+    @staging_project = Project.find_by(name: params[:staging_project])
+    raise ActiveRecord::RecordNotFound unless @staging_project
```

Comment:
[nitpick] This was already there but I think we should use render_error here instead.

Reply:
in the api, check https://github.com/vpereira/open-build-service/blob/d7d2e11f642cf8148573b5d12b5a607d4058c182/src/api/app/controllers/application_controller.rb#L239 

we have something like

```
 rescue_from Backend::NotFoundError, ActiveRecord::RecordNotFound do |exception|
    render_error message: exception.message, status: 404, errorcode: 'not_found'
  end 
```

which, IMO,  does the trick

Reply:
Ok, fine for me then :+1: